uefi-pi/src/pi/fv.rs

```diff
@@ -62,6 +76,16 @@ impl FirmwareVolumeHeader {
     pub fn as_bytes(&self) -> &[u8] {
         unsafe { &*slice_from_raw_parts(self as *const Self as *const u8, size_of::<Self>()) }
     }
+
+    // Calculate and update the checksum of the FirmwareVolumeHeader
+    pub fn update_checksum(&mut self) {
+        self.checksum = u16::MAX - sum16(self.as_bytes()) + 1;
+    }
```

Comment:
What worries me is that this method reads the struct as bytes and then modifies that struct to write the checksum back into it. That implies that if there is _already_ a checksum set, the new checksum calculation will be wrong.

A simple fix would be to set `fv_header.checksum = 0` before calling `update_checksum()` I think as that would effectively exclude the checksum from the calculation of... the checksum :smile:

```suggestion
    pub fn update_checksum(&mut self) {
        // Exclude any existing checksum from the checksum calculation
        self.checksum = 0;
        
        self.checksum = u16::MAX - sum16(self.as_bytes()) + 1;
    }
```

That would also make the method idempotent (it currently isn't fwics).

Reply:
Right, it was missed, thanks for your suggestion!😊